### PR TITLE
setup CI test runner and fix broken tests

### PR DESCRIPTION
**Pull request by mschneider**
includes #22 & #23

### CODE REVIEW

**Comment by yihau** [comment]
ah! seems I missed some other cases haha. thank you for doing this for me 🚀

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+            dex/target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: test
+        run: |
+          cargo test --manifest-path Cargo.toml --all
+          cargo test --manifest-path dex/Cargo.toml --all
```
Comment (SpaceMonkeyForever) [inline .github/workflows/test.yml:31]: Why do we need this second line?

**Review by SpaceMonkeyForever — COMMENTED** [review]
change makes sense and fixes the test, verified locally and github output is correct. The only question I have is why we need that second line in test.yml (see comment)

File: .github/workflows/test.yml (hunk 1 above)
Comment (mschneider) [reply .github/workflows/test.yml:31]: the root cargo toml excludes the program:

```
[workspace]
members = [
    "assert-owner",
    "common",
    "dex/crank",
    "dex/permissioned",
]
exclude = [
    "dex",
]
```

**Review by riordanp — APPROVED** [review]
looks good to me, caching seems to work well on the action

**Pull request merged**